clang/test/CAS/fcache-compile-job-dependency-file.c

```diff
@@ -31,11 +31,12 @@
 // RUN: %clang -cc1 -triple x86_64-apple-macos11 \
 // RUN:   -fcas-path %t/cas -faction-cache-path %t/cache -fcas-fs @%t/casid -fcache-compile-job \
 // RUN:   -Rcompile-job-cache %t/main.c -emit-obj -o %t/output.o -isystem %t/sys \
-// RUN:   -dependency-file %t/deps3.d -MT other1 -MT other2 -MP 2>&1 \
+// RUN:   -dependency-file %t/deps3.d -MT other1 -MT other2 -MP -fdepfile-entry=extra-depfile.json 2>&1 \
 // RUN:   | FileCheck %s --check-prefix=CACHE-HIT
 
 // RUN: FileCheck %s --input-file=%t/deps3.d --check-prefix=DEPS_OTHER
 // DEPS_OTHER: other1 other2:
+// DEPS_OTHER: extra-depfile.json
```

Comment:
I think this may be broken if you do the opposite change: add `-fdepfile-entry` to the first invocation, and remove it in the second invocation.  The reason is that `CASDependencyCollector` inherits from `DependencyFileGenerator` and that class adds these deps in its constructor.

Reply:
Oh, good catch! I'll take a look.

Reply:
See updated commit.